packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -276,6 +313,26 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
         ],
       };
     }
+  } else if (stateP.controlState === 'PREPARE_COMPATIBLE_MIGRATION') {
+    const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+    if (Either.isRight(res)) {
+      return {
+        ...stateP,
+        controlState: 'OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT',
+      };
+    } else if (Either.isLeft(res)) {
+      // We assume that the alias was already deleted by another Kibana instance
```

Comment:
nit: we agreed it's out of scope, but I might be worth adding a note here that we don't handle the case of multiple, different newer versions doing a migration concurrently. For that we'd have to add a `PREPARE_COMPATIBLE_MIGRATION_CONFLICT` step similar to `MARK_VERSION_INDEX_READY_CONFLICT`

Reply:
Are there possible side effects of not controlling this case?
E.g. imagine a scenario where:
* indices are in 8.6.0
* we're starting 8.8.0 and 8.7.0 in parallel
* mappings have not changed in either of the new releases, but we have transform functions that MUTATE some values of a property, and they do it differently in 8.8.0 vs 8.7.0.

Could we have a lost update problem (caused by 8.7.0 overriding 8.8.0's changes) in this case?